src/serializer.ts

```diff
@@ -62,12 +70,24 @@ class Serializer {
    * @returns {Object} The Payment as JSON.
    */
   private static paymentToJSON(payment: Payment): object | undefined {
-    const json = {
-      TransactionType: "Payment",
-      Destination: payment.getDestination(),
-      Amount: {}
+    const json: PaymentJSON = {
+      Amount: {},
+      Destination: "",
+      TransactionType: "Payment"
     };
 
+    // If an x-address was able to be decoded, add the components to the json.
+    const decodedXAddress = Utils.decodeXAddress(payment.getDestination());
+    if (!decodedXAddress) {
+      json.Destination = payment.getDestination();
+      delete json.DestinationTag;
+    } else {
+      json.Destination = decodedXAddress.address;
+      if (decodedXAddress.tag) {
```

Comment:
Since `0` is a valid `tag`, I think this should be:
```
      if (decodedXAddress.tag !== undefined) {
```

Reply:
Done